### DISCUSSION TITLE
Prisma Types for Arrays

### DISCUSSION THREAD

**Discussion by DavidJakszta** [prisma/prisma#19623] [Q&A] [answered]
So i've been using the prisma generated types for typescript but now we added a List to the schema and the type i used to use doenst work anymore
schema:
```prisma
model Post {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now()) @map(name: "created_at")
  updatedAt DateTime @default(now()) @map(name: "updated_at")
  userId    String
  content   String
  likeCount Int?     @default(0)
  published Boolean  @default(false)

  user    User      @relation(fields: [userId], references: [id], onDelete: Cascade)
  Comment Comment[]
  Like    Like[]

  @@map(name: "posts")
}
```

type:
```js
  export type PostUncheckedCreateInput = {
    id?: string
    createdAt?: Date | string
    updatedAt?: Date | string
    userId: string
    content: string
    likeCount?: number | null
    published?: boolean
    Comment?: CommentUncheckedCreateNestedManyWithoutPostInput
    Like?: LikeUncheckedCreateNestedManyWithoutPostInput
  }
```

so for Read Purposes this probably isnt the best type but it worked. And now i cant use them in code anymore. I would basically just like to get the Commentlist length but this type does not work

Now my question: Does Prisma Generate Types where the Comment and Like lists are arrays? or how would i go about that

**Comment by nurul3101** [comment, marked as answer]
Hello @wtfmynamefits 👋 

In your case, the `Comment` and `Like` fields in your `Post` model are defined as arrays, which means that they are one-to-many relations. 

However, the specific type you're using (`PostUncheckedCreateInput`) is meant for creating new `Post` instances and bypassing some checks for nullable fields. This type does not include the related fields as arrays because at the creation time, there might not be any related `Comment` or `Like` instances.

To access the `Comment` and `Like` lists as arrays, you need to use a different type, typically generated by Prisma, that includes these relations. You might want to look for types like `PostWhereInput` in your generated Prisma client.

I would also recommend you to have a look at this [section](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type) in our docs which discusses the same use case.

**Reply by nurul3101** [reply]
@wtfmynamefits Did you had a chance to look at my last response.

If this answers your question, please feel free to mark this discussion as answered. 🙏

**Reply by DavidJakszta** [reply]
no i dont understand how to do this without writing any of my own types. Can you make an example?

**Reply by DavidJakszta** [reply]
nevermind i understand now. Im kinda sad that thats alot of code for something that simple and it cant be fixed easily when i generate again after adding new types
